syntax = "proto2";

option cc_enable_arenas = true;

package c2017.shooter;

message ShooterInput {
  required double encoder_position = 1;
}

message ShooterOutput {
  required double voltage = 1;
}

message ShooterStatus {
  required double observed_velocity = 1; // Positive velocity = forwards
}

message ShooterGoal {
  enum ShotMode {
    FENDER = 0;
    HOPPER = 1;
  }
  required ShotMode goal_mode = 1;
  required double goal_velocity = 2;
}
